Turn cookbook recepies into examples

romanowski opened the pull request.

romanowski committed:
Turn cookbook recepies into examples

alexarchambault commented on docs_checker/check.scala:
```diff
@@ -58,68 +60,105 @@ def parse(content: Seq[String], currentCommands: Seq[Commands], context: Context
+
+  // remove empty space at begining of all files
+  if dest.nonEmpty then
+    val header = s"File was generated from based on ${file}, do not edit manually!"
```
NIT
```suggestion
    val header = s"This file was generated from $file, do not edit manually!"
```

alexarchambault commented on docs_checker/check.scala:
```diff
@@ -1,5 +1,6 @@
 // using scala 3.0.2
 // using "org.scalameta::munit:0.7.29"
+// using com.lihaoyi:ammonite-ops_2.13:2.4.0
```
Wouldn't [os-lib](https://github.com/com-lihaoyi/os-lib) (`com.lihaoyi::os-lib:0.7.5`) be enough? Ammonite-ops is more or less deprecated IIRC.

romanowski replied:
I just pick a random lib, I will change that

romanowski committed:
Add more logging what was not commite and run scala-cli directly

romanowski committed:
Migrate to os.lib

romanowski committed:
Improve cookbooks

- add `fail` flag to make sure that command fail
- skip frontmatter in generated readmefiles
- migrate to os-lib

lwronski commented on docs_checker/check.scala:
```diff
@@ -58,68 +59,118 @@ def parse(content: Seq[String], currentCommands: Seq[Commands], context: Context
       Seq(TestCase(path, Some(e)))
 
-def checkFile(file: Path) =
+def ammPath(p: Path) = os.Path(p.toAbsolutePath)
```
I think, this method can be removed if you migrate from `Java - Paths.*` to `os.lib`.

lwronski commented on docs_checker/check.scala:
```diff
@@ -58,68 +59,118 @@ def parse(content: Seq[String], currentCommands: Seq[Commands], context: Context
+val fakeLineMarker = "//fakeline"
+
+def checkFile(file: Path, dest: Option[Path]) =
   val content  = Files.lines(file).iterator.asScala.toList
```
```suggestion
  val content  = os.read.lines(file)
```
but, you have to replace `Path` to `os.Path`

romanowski committed:
Update to master
